src/models/shelf/filter.ts

```diff
@@ -114,3 +114,31 @@ export function containsFilter(filters: Array<RangeFilter | OneOfFilter>, target
   }
   return false;
 }
+
+export function getModifyTimeUnitFunction(timeUnit: TimeUnit, domain: number[]) {
```

Comment:
This logic is only called in reducer so it could just be in the reducer.
(I suggest model only if it is called directly in component.)